### DISCUSSION TITLE
Log buffers for ranges shown by `git fetch`

### DISCUSSION THREAD

**Discussion by peniblec** [magit/magit#4692] [Ideas]
Howdy!

I have a routine for "catching up" to a repository's activity that consists in

1. <kbd>f u</kbd>
2. <kbd>$</kbd>
3. see if branches I'm interested in have been updated,
4. for each such branch,
    1. move to its `<old>..<new> <from> -> <to>` line,
    2. <kbd>M-d M-d</kbd> to grab `<old>..<new>`,
    3. <kbd>l o C-y RET</kbd>.

This is fine, and not even necessary for branches I have checked out since I can just use the `Unpulled from UPSTREAM/BRANCH` section in the status buffer; still, I would often like to review new commits for branches that I don't have checked out.

So off the top of my head, I can think of a couple of ways to make this process slightly easier:

1. Teach `magit-log-other` to recognize `<old>..<new>` ranges at point and offer them in the minibuffer's "future history" (<kbd>M-n</kbd>),

2. Post-process `git fetch`'s output and buttonize the ranges shown in the process buffer, so that <kbd>RET</kbd> on a button spawns a log buffer for this range,

3. Post-process `git fetch`'s output and spawn a log buffer with all those ranges, as if the user had copied them one by one, then invoked <kbd>l o C-y , C-y M-y , C-y M-y M-y… RET</kbd>).

My two questions would be:

1. Is there already a better way to achieve what I am doing? (Spawn a log buffer for an `<old>..<new>` range in the output of `git fetch`, for arbitrary branches that might not be checked out)

   (<kbd>l o origin/BRANCH@{1}..origin/BRANCH</kbd> works, but it's not an improvement over my current routine)

2. Do any of these suggestions sound like interesting additions?

Suggestion 1 could prove useful in any place where dotted ranges show up, and it could also be applied to <kbd>d r</kbd> as well as <kbd>l o</kbd> (right now `(magit-diff--dwim)` returns `nil` when point is on a range). The biggest issue I see is that it would only be intuitive for users who know about <kbd>M-n</kbd>.

Suggestion 2 sounds like the most user-friendly, if only because the button face is an obvious visual cue, but Magit does not do a lot of post-processing on command output by default AFAIK, so I don't know if this is desirable.  Maybe as an opt-in feature, gated behind a user option? (`magit-process-finish-buttonize-ranges`?)  But then some users might prefer the button to show a diff rather than a log…

Suggestion 3 sounds kind of heavy-handed to me 🤷

WDYT?  I don't mind taking a crack at any of these ideas, if they sound like improvements.

**Comment by tarsius** [comment]
Suggestion 1 and 2 sound good to me too. 2 is more costly, so that should probably be optional. So users would have to "know" that there is an option to enable it just like they have to know about `M-n` for 1. I would suggest you try implementing 1 first.
